rocks-bin/src/run_lua.rs

```diff
@@ -55,7 +55,17 @@ pub async fn run_lua(run_lua: RunLua, config: Config) -> Result<()> {
         }
     }
     let tree = config.tree(lua_version)?;
-    let paths = Paths::from_tree(tree)?;
+
+    let paths = if let Some(project) = project {
+        let mut paths = Paths::new(tree)?;
+
+        paths.prepend(&Paths::new(project.tree(&config)?)?);
+
+        paths
```

Comment:
question: If I understand this correctly, this makes the user-installed packages available on the paths.
Is that the behaviour we want? I think we should only include the local project tree.

Reply:
I do think we want to include all packages. The ones in the project have higher precedence by being earlier in the path.

I'd expect `rocks lua` to be a wrapper around Lua with all the relevant packages available to the REPL. The ones I installed in the user-wide tree also count as relevant, I'd say :p

Reply:
scenario:

- I have package `foo` installed in my user tree
- My project depends on `foo`, but it's not listed in the rocks.toml dependencies
- I'm too dumb to write a test suite and I only use `rocks lua` to test my package manually
- My manual tests succeed
- I publish my package

I think we should default to not including all packages, but perhaps could add an `--impure` flag that changes the behaviour.
But I really can't think of a scenario where you'd actually want that.

Reply:
A `--impure` flag sounds nice, but shouldn't the user be running their code through the (hopefully) upcoming `rocks run` anyway, which will be a pure environment?

Another option would be to differentiate script-running mode (pure by default) and REPL mode (impure by default), which I think sounds pretty sensible without being too "magical". Wdyt?

Reply:
Sure, that sounds fine to me.

Reply:
Let me make a tracking issue for that.